src/overlays/actors/ovl_Obj_Warpstone/z_obj_warpstone.c

```diff
@@ -28,37 +30,157 @@ const ActorInit Obj_Warpstone_InitVars = {
+        }
+    }
+    return true;
+}
+
+s32 ObjWarpstone_OpenedIdle(ObjWarpstone* this, GlobalContext* globalCtx) {
+    /*You can save your progress and quit here.*/
+    this->dyna.actor.textId = 0xC01;
+    return false;
+}
+
+void ObjWarpstone_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjWarpstone* this = THIS;
+    s32 pad;
+
+    if (this->isTalking) {
+        if (func_800B867C(&this->dyna.actor, globalCtx) != 0) {
+            this->isTalking = false;
+        } else if ((func_80152498(&globalCtx->msgCtx) == 4) && (func_80147624(globalCtx))) {
+            if (globalCtx->msgCtx.choiceIndex != 0) {
+                func_8019F208();
+                globalCtx->msgCtx.unk11F22 = 0x4D;
+                globalCtx->msgCtx.unk120D6 = 0;
+                globalCtx->msgCtx.unk120D4 = 0;
+                gSaveContext.owlSaveLocation = GET_OWL_ID(this);
+            } else {
+                func_801477B4(globalCtx);
+            }
+        }
+    } else if (func_800B84D0(&this->dyna.actor, globalCtx)) {
+        this->isTalking = true;
+    } else if (!this->actionFunc(this, globalCtx)) {
+        func_800B863C(&this->dyna.actor, globalCtx);
+    }
+    Collider_ResetCylinderAC(globalCtx, &this->collider.base);
+    Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void ObjWarpstone_Draw(Actor* thisx, GlobalContext* globalCtx2) {
+    ObjWarpstone* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+
+    func_800BDFC0(globalCtx, sOwlStatueDLs[this->modelIndex]);
+    if (this->dyna.actor.home.rot.x != 0) {
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        SysMatrix_InsertTranslation(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y + 34.0f,
+                                    this->dyna.actor.world.pos.z, MTXMODE_NEW);
+        SysMatrix_InsertMatrix(&globalCtx->mf_187FC, MTXMODE_APPLY);
+        SysMatrix_InsertTranslation(0.0f, 0.0f, 30.0f, MTXMODE_APPLY);
+        Matrix_Scale(this->dyna.actor.velocity.x, this->dyna.actor.velocity.x, this->dyna.actor.velocity.x,
+                     MTXMODE_APPLY);
+        SysMatrix_StatePush();
+        gDPPipeSync(POLY_XLU_DISP++);
+        gDPSetPrimColor(POLY_XLU_DISP++, 128, 128, 255, 255, 200, this->dyna.actor.home.rot.x);
+        gDPSetEnvColor(POLY_XLU_DISP++, 100, 200, 0, 255);
+        SysMatrix_InsertZRotation_f((((globalCtx->gameplayFrames * 1500) & 0xFFFF) * M_PI) / 32768.0f, MTXMODE_APPLY);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, D_04023210);
+        SysMatrix_StatePop();
+        SysMatrix_InsertZRotation_f((~((globalCtx->gameplayFrames * 1200) & 0xFFFF) * M_PI) / 32768.0f, MTXMODE_APPLY);
```

Comment:
```suggestion
        SysMatrix_InsertZRotation_f((((globalCtx->gameplayFrames * 1500) & 0xFFFF) * M_PI) / (f32)0x8000, MTXMODE_APPLY);
        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
        gSPDisplayList(POLY_XLU_DISP++, D_04023210);
        SysMatrix_StatePop();
        SysMatrix_InsertZRotation_f((~((globalCtx->gameplayFrames * 1200) & 0xFFFF) * M_PI) / (f32)0x8000, MTXMODE_APPLY);
```
There's a chance the `(f32)` cast isn't needed

Reply:
f32 cast is not needed. 0x8000 works for both. I just pushed that change, so I'm marking this as resolved.